src/dropdown/Dropdown.js

```diff
@@ -199,9 +192,20 @@ export default class Dropdown extends React.Component {
   }) => {
 
     const Option = optionComponent;
-    const propertyName = this.props.groupByKey;
+    const { groupByKey } = this.props;
+    const groupedOptions = options.reduce((acc, o, i) => {
+      if (i === 0 || o[groupByKey] !== options[i - 1][groupByKey]) {
+        return [...acc, { optionGroupTitle: o[groupByKey] || '', optionGroup: [o] }];
+      } else {
+        const lastGroup = acc.slice(-1)[0];
+        return [...acc.slice(0, acc.length - 1), {
+          optionGroupTitle: lastGroup.optionGroupTitle,
+          optionGroup: [...lastGroup.optionGroup, o]
+        }];
+      }
+    }, []);
```

Comment:
I was relying on the fact that options are already grouped and sorted and we don't really need to group & sort again. We can just iterate on options and check the previous one, pushing options to the current group 'til the group stay the same, or creating a new group when we hit the first occurrence of a new group